pkg/asset/asset.go

```diff
@@ -55,3 +56,39 @@ func PersistToFile(asset WritableAsset, directory string) error {
 	}
 	return nil
 }
+
+// DeleteAssetFromDisk removes all the files for asset from disk.
+func DeleteAssetFromDisk(asset WritableAsset, directory string) error {
+	for _, f := range asset.Files() {
+		path := filepath.Join(directory, f.Filename)
+		if err := os.Remove(path); err != nil && !os.IsNotExist(err) {
+			return errors.Wrapf(err, "failed to remove file %q", f.Filename)
+		}
+
+		dir := filepath.Dir(path)
+		ok, err := isDirEmpty(dir)
+		if err != nil && !os.IsNotExist(err) {
+			return errors.Wrapf(err, "failed to read parent directory for file %q", f.Filename)
+		}
+		if ok {
+			if err := os.Remove(dir); err != nil {
+				return errors.Wrapf(err, "failed to remove dir of file %q", f.Filename)
```

Comment:
No need to include the path here, it's part of [the returned `*PathError`][1].

```console
$ cat test.go 
package main

import (
	"fmt"
	"os"

	"github.com/pkg/errors"
)

func Delete(path string) (err error) {
	return errors.Wrap(os.Remove(path), "delete empty directory")
}

func main() {
	fmt.Println(Delete("does-not-exist"))
}
```

[1]: https://golang.org/pkg/os/#Remove